#! /bin/sh
patch -p1 -f $* < $0
exit $?

Fix a bug in the W32 plugin directory handling.

2006-01-09  Werner Koch  <wk@g10code.com>

	* src/common/utils.c (w32_get_locale_dir): Removed.
	(w32_get_module_dir): New.
	(get_local_dir): Use new function.
	(get_plugin_dir): Ditto.


diff -ru sylpheed-claws.orig/src/common/utils.c sylpheed-claws/src/common/utils.c
--- sylpheed-claws.orig/src/common/utils.c	2006-01-05 16:36:51.000000000 +0100
+++ sylpheed-claws/src/common/utils.c	2006-01-09 18:59:18.000000000 +0100
@@ -1980,24 +1980,28 @@
     return -1;
 }
 
-/* Return a malloced string with the locale dir.  Return NULL on
-   error. */
-static char *w32_get_locale_dir(void)
-{
-        char name[MAX_PATH+35];
-        char *p;
-
-        if ( !GetModuleFileNameA (0, name, sizeof (name)-30) )
-                return NULL;
-        p = strrchr (name, '\\');
-        if (!p)
-        {
-                *name = '\\';
-                p = name;
+/* Returns a static string with the directroy from which the module
+   has been loaded.  Returns an empty string on error. */
+static char *w32_get_module_dir(void)
+{
+        static char *moddir;
+
+        if (!moddir) {
+                char name[MAX_PATH+10];
+                char *p;
+
+                if ( !GetModuleFileNameA (0, name, sizeof (name)-10) )
+                        *name = 0;
+                else {
+                        p = strrchr (name, '\\');
+                        if (p)
+                                *p = 0;
+                        else
+                                *name = 0;
+                }
+                moddir = g_strdup (name);
         }
-        p++;
-        strcpy (p, "\\share\\locale");
-        return g_strdup (name);
+        return moddir;
 }
 #endif /* G_OS_WIN32 */
 
@@ -2008,7 +2012,8 @@
 
 #ifdef G_OS_WIN32
 	if (!loc_dir)
-                loc_dir = w32_get_locale_dir();
+		loc_dir = g_strconcat(w32_get_module_dir(), G_DIR_SEPARATOR_S,
+                                      "\\share\\locale", NULL);
 #endif
 	if (!loc_dir)
 		loc_dir = LOCALEDIR;
@@ -2133,7 +2138,7 @@
 	static gchar *plugin_dir = NULL;
 
 	if (!plugin_dir)
-                plugin_dir = g_strconcat(get_locale_dir(),
+                plugin_dir = g_strconcat(w32_get_module_dir(),
                                          "\\lib\\sylpheed-claws\\plugins\\",
                                          NULL);
 	return plugin_dir;


